src/main/java/org/commcare/formplayer/sandbox/SqlStorageWrapper.java

```diff
@@ -0,0 +1,305 @@
+package org.commcare.formplayer.sandbox;
+
+import org.commcare.formplayer.services.ConnectionHandler;
+import org.javarosa.core.model.condition.RequestAbandonedException;
+import org.javarosa.core.services.storage.EntityFilter;
+import org.javarosa.core.services.storage.IStorageUtilityIndexed;
+import org.javarosa.core.services.storage.Persistable;
+import org.javarosa.core.util.InvalidIndexException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Vector;
+import java.util.concurrent.Callable;
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.datadog.DatadogMeterRegistry;
+
+/**
+ * @author $|-|!˅@M
+ */
+public class SqlStorageWrapper<T extends Persistable>
+        implements IStorageUtilityIndexed<T>, Iterable<T> {
+
+    private SqlStorage<T> sqliteStorage;
+    private SqlStorage<T> postgresStorage;
+    private DatadogMeterRegistry meterRegistry;
+
+    public SqlStorageWrapper(ConnectionHandler sqliteConnection, ConnectionHandler postgresConnection,
+                             Class<T> prototype, String tableName, DatadogMeterRegistry meterRegistry) {
+        sqliteStorage = new SqlStorage<T>(sqliteConnection, prototype, tableName);
+        postgresStorage = new SqlStorage<T>(postgresConnection, prototype, tableName);
+        this.meterRegistry = meterRegistry;
+    }
+
+    private void compareRunnable(String tag, Runnable sqliteOperation, Runnable postgresOperation) {
+        Timer sqliteTimer = meterRegistry.timer("timer.sqlite." + tag);
+        sqliteTimer.record(sqliteOperation);
+
+        Timer postgresTimer = meterRegistry.timer("timer.postgres." + tag);
+        postgresTimer.record(postgresOperation);
+    }
+
+    private <RESULT> RESULT compareCallable(String tag, Callable<RESULT> sqliteOperation, Callable<RESULT> postgresOperation) {
+        RESULT result = null;
+        try {
+            long start = System.currentTimeMillis();
+            result = sqliteOperation.call();
+            long sqliteTime = System.currentTimeMillis() - start;
+
+            start = System.currentTimeMillis();
+            postgresOperation.call();
+            long postgresTime = System.currentTimeMillis() - start;
```

Comment:
Is there anyway to do these 2 asynchronously/simultaneously? 

A potential way we can measure the performance is to see if the overall request is slower. This might be important because even though individual postgres queries might be only subseconds slower than their sqlite counterpart, if we do multiple in the same request it adds up. And in this case, the overall request might be twice as slow if we're doing one request then the other.